test/test_core.py

```diff
@@ -4131,6 +4133,54 @@ def test_dlfcn_asyncify(self):
       '''
     self.do_run(src, 'before sleep\nafter sleep\n42\n')
 
+  @needs_dylink
+  def test_dlfcn_rtld_local(self):
+    create_file('liba.c', r'''
+      #include <stdio.h>
+
+      void func_b();
+
+      void func_a() {
+        printf("func_a\n");
+        // Call a function from a dependent DSO. This symbol should
+        // be available here even though liba itself is loaded with RTLD_LOCAL.
+        func_b();
+      }
+      ''')
+
+    create_file('libb.c', r'''
+      #include <stdio.h>
+
+      void func_b() {
+        printf("func_b\n");
+      }
+    ''')
+
+    self.build_dlfcn_lib('libb.c', outfile='libb.so')
+    self.build_dlfcn_lib('liba.c', outfile='liba.so', emcc_args=['libb.so'])
+
+    self.prep_dlfcn_main(['liba.so', 'libb.so', '-L.'])
+    create_file('main.c', r'''
+      #include <assert.h>
+      #include <dlfcn.h>
+      #include <stdio.h>
+
+      int main() {
+        printf("main\n");
+        void* handle = dlopen("liba.so", RTLD_NOW|RTLD_LOCAL);
+        assert(handle);
+
+        void (*f)();
+        f = dlsym(handle, "func_a");
+        assert(f);
+        f();
+        printf("done\n");
+        return 0;
+      }
```

Comment:
What do you mean by "readable" here?  Module b does not link against module b so I think it would be a compile error to try to use `func_a` directly in the source code.

Or are  you talking about module b doing a dlopen maybe?

One think we should check is that neither `func_a` nor `func_b` are available in the default / global symbol lookup space (`dlopen(RTLD_DEFAULT)`).

Reply:
I'm looking for a test of the 'locality'. Reading the docs, it says that `RTLD_LOCAL` means a library's symbols are not available in later libraries loaded. So failing to `dlsym` from them could be tested? Or does that only affect global symbols and not `dlsym` calls?

Reply:
(Basically, a test that works with `RTLD_GLOBAL` but fails with `RTLD_LOCAL`, and we can verify that it fails.)

Reply:
Done, ptal